design/controlling-agent-updates.md

```diff
@@ -0,0 +1,66 @@
+# Controlling agent updates
+
+The agent is considered an extension of the service.
+As such, we have an auto-update mechanism to ensure that new features can rely on the agent support they require.
+Conceptually, it's no different from binaries on the app tier or JavaScript served to the web frontend.
+
+For instance, tasks express a minimum agent version required.
+If a selected agent doesn't meet that requirement, the agent is first asked to update itself.
+Other ways an agent can be updated include using a feature which demands an evergreen agent (YAML pipelines, for instance) or clicking the "upgrade all agents" button in a pool.
+
+For some highly-regulated customers, this kind of evergreen infrastructure is at odds with their compliance needs.
+Such customers today are doing things like ACLing the agent's installation directory so it can't update itself or recompiling the agent without update support.
+This leaves us in a situation where no one is happy:
+we can't service the product correctly, and the customer has to maintain custom configuration / infrastructure to meet their compliance needs.
+
+## Solution
+
+_See below for other solutions considered._
+
+We will add a mode to the project-level pool which puts it into "no update" mode.
+In this mode, jobs will use the minimum agent version as a demand.
+If no agent can satisfy the demand, the job will fail instead of forcing an agent update.
+
+The error message must clearly indicate what feature or task demanded agent update.
+It must also indicate that the customer opted into this behavior.
+This way, the customer can decide whether to update their agents or edit the pipeline to remove the offending feature.
+For example:
+> This pool doesn't allow agent updates.
+> An administrator can alter this setting on the project pool page.
+> Task `MyTask` version 1.2.3 requires agent version 2.175.1, which could not be satisfied.
+
+Manual agent updates (i.e. clicking the button in the web) should still send update requests as normal.
+
+## Potential problems & solutions
+
+**If a small number of agents have been updated, then all jobs will funnel to that agent instead of spreading out across the pool.**
```

Comment:
But if the manual update action, and the setting, are both at the pool level, we shouldn't get into this situation, right?

Reply:
I'll clarify in the doc. You can get into this state by manually updating a few agents, or clicking the "update this agent" button in the UI (yep, you can do that at the pool level and at the individual agent level).